### PR TITLE
[feat] Keypress Functionality

### LINKED ISSUE #93: Add keypress functionality to Auth and Navigation
https://linear.app/trap-garden/issue/TG-59/add-keypress-functionality-to-auth-and-navigation

### PR DESCRIPTION
**Pull request by torinnayak1**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

For the login, signup, onboarding, and add details pages, instead of pressing submit (or "Let's Grow") for onboarding, the user can press enter, with the same functionality. Also, if the user is viewing the nav bar, they can press escape to get out of it.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

The page.tsx for login, signup, onboarding, add-details, navcolumn can be reviewed in any order

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

cleaning up the onboarding steps so that I don't add a boof extra argument maybe

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

Rachel's work about the forgot password flow and anything related to that sign up may be in conflict.

CC: @kylezryr

### CODE REVIEW

File: app/resources/page.tsx
Comment (kylezryr) [inline app/resources/page.tsx:1]: we can probably remove this file from the PR. after the resources PR is merged you can rebase and this should disappear, so we can get back to this later!

File: app/resources/styles.ts
Comment (kylezryr) [inline app/resources/styles.ts:1]: same thing abt rebasing for this file

File: app/(auth)/login/page.tsx [hunk 1]
```diff
@@ -54,6 +54,23 @@ export default function Login() {
...
+        }
+      }
+    };
+
+    //add listener for keydown events
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [email, password]);
```
Comment (kylezryr) [inline app/(auth)/login/page.tsx]: we should add `handleLogin` as a dependency here too to avoid lint warnings

File: components/FAQDropdown/index.tsx
Comment (kylezryr) [inline components/FAQDropdown/index.tsx:1]: this file also to be removed after rebase

File: app/(auth)/signup/page.tsx [hunk 2]
```diff
@@ -81,6 +81,28 @@ export default function SignUp() {
...
+        }
+      }
+    };
+
+    //add listener for keydown events
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  });
```
Comment (kylezryr) [inline app/(auth)/signup/page.tsx]: let's also add the relevant variables to a dependency here, so `email, password, passwordsMatch, isPasswordComplexityMet, handleSignUp`

File: app/add-details/page.tsx [hunk 3]
```diff
@@ -108,6 +108,21 @@ export default function Home() {
...
+        handleSubmit();
+      }
+    };
+
+    //add listener for keydown events
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  });
```
Comment (kylezryr) [inline app/add-details/page.tsx]: add `handleSubmit` as a dependency

File: app/onboarding/page.tsx [hunk 4]
```diff
@@ -117,10 +118,24 @@ const ReviewPage = ({
...
+        handleSubmit();
+      }
+    };
 
+    //add listener for keydown events
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  });
```
Comment (kylezryr) [inline app/onboarding/page.tsx]: add `currStep, handleSubmit` as dependencies

**Review by kylezryr — CHANGES_REQUESTED** [review]
looks good overall! just to add the dependencies for checking conditions in each of the `useEffects` and to also rebase this branch after the resources page PR is merged so that we don't see the changes from that PR! great work

**Review by kylezryr — APPROVED** [review]
changes look good! i just removed the other files from this PR since they shouldn't be here anyways, they'll be added in the resource page PR

**Review by SashankBalusu — APPROVED** [review]
quite literally the best pr ive ever seen
goat shit
